pkg/auth/authenticator/password/basicauthpassword/basicauthpassword.go

```diff
@@ -80,6 +94,11 @@ func (a *Authenticator) AuthenticatePassword(username, password string) (user.In
 
 	resp, err := a.client.Do(req)
 	if err != nil {
+		if _, ok := err.(*RedirectAttemptedError); !ok {
+			// If this was a redirect, treat it as a simple
+			// authentication failure, not an error
```

Comment:
If the idp is not returning a 401 or 200, it is misconfigured, and I think the user should get a 500.